Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -883,16 +899,73 @@ GRefPtr<GstElement> createOptionalParserForFormat([[maybe_unused]] GstBin* bin,
             // defines several extensions to the original AAC, also widely used.
             // Not to be confused with the MP4 file format, which is a container format, not an audio stream format,
             // and can incidentally contain MPEG-4 audio.
-            // Make sure each frame has a header, as mandated by the frame() syntax element
-            // specified in Section 2.4.1.2 of ISO 11172-3. Without specifying a stream-format,
-            // aacparse will fallback to raw, which might confuse some decoders (avdec) leading to
-            // garbled audio rendering.
-            GUniqueOutPtr<GError> error;
-            result = gst_parse_bin_from_description("aacparse ! capsfilter caps=\"audio/mpeg, stream-format=(string)adts\"", TRUE, &error.outPtr());
-            if (result)
-                gst_object_set_name(GST_OBJECT_CAST(result.get()), parserName.ascii().data());
-            else
-                GST_WARNING_OBJECT(bin, "Unable to create AAC parser: %s", error->message);
+            // Replace explicit signalling of SBR (HE AAC) and PS (HE AACv2) with implicit
+            // signalling (i.e. let the decoder figure out what AAC extensions are used).
+            // Unfortunately it is not uncommon for AAC streams to explicitly signal extensions they
+            // don't actually use, which is known to cause problems in avdec_aac.
+
+            result = makeGStreamerElement("aacparse"_s, parserName);
+            if (!result)
+                break;
+
+            gst_pad_add_probe(demuxerSrcPad.get(), GST_PAD_PROBE_TYPE_EVENT_DOWNSTREAM, reinterpret_cast<GstPadProbeCallback>(+[]([[maybe_unused]] GstPad* pad, GstPadProbeInfo* info, gpointer) -> GstPadProbeReturn {
```

Comment:
nit: this is too big of a lambda and would be better off as a separate static function. You could call it `aacSbrForceImplicitSignalling` or something of the like.